osvimdriver/openstack/neutron/driver.py

```diff
@@ -45,5 +71,53 @@ def get_subnet_by_id(self, subnet_id):
             raise ValueError('subnet_id must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving subnet with id %s', subnet_id)
-        result = neutron_client.show_subnet(subnet_id)
-        return result['subnet']
+        try:
+            external_request_id = str(uuid.uuid4())
+            driver_request_id  = str(uuid.uuid4())
```

Comment:
driver_request_id is specific to a lifecycle, Please pass this value from the calling function.

Reply:
I have done changes.